### PR TITLE
Recommend use of ln1p/log1p to avoid negative scores

### PR DESCRIPTION
**Pull request by AthenaEryma**
As negative scores will now cause an error, and it is easy to
accidentally produce negative scores with some of the built-in modifiers
(especially `ln` and `log`), this adjusts the documentation to more
strongly recommend the use of `ln1p` and `log1p` instead.

Also corrects some awkward formatting on the note sections following the
table.

Relates to https://github.com/elastic/elasticsearch/issues/41509

### CODE REVIEW

File: docs/reference/query-dsl/function-score-query.asciidoc [hunk 1]
```diff
@@ -321,14 +325,17 @@ There are a number of options for the `field_value_factor` function:
     Value used if the document doesn't have that field. The modifier
     and factor are still applied to it as though it were read from the document.
 
+IMPORTANT: Scores produced by the `field_value_score` function must be
```
Comment (jtibshirani) [inline docs/reference/query-dsl/function-score-query.asciidoc]: This is just personal preference (feel free to ignore), but this got me thinking when we use `IMPORTANT:` vs. `NOTE:`. This and the following paragraph seem more like notes to me as it's a pretty routine description of a limitation.

File: docs/reference/query-dsl/function-score-query.asciidoc [hunk 2]
```diff
@@ -305,10 +305,14 @@ There are a number of options for the `field_value_factor` function:
 | Modifier | Meaning
 
 | `none` | Do not apply any multiplier to the field value
-| `log` | Take the https://en.wikipedia.org/wiki/Common_logarithm[common logarithm] of the field value
+| `log` | Take the https://en.wikipedia.org/wiki/Common_logarithm[common logarithm] of the field value.
+          Because this function will return a negative value and cause an error  if used on values
```
Comment (jtibshirani) [inline docs/reference/query-dsl/function-score-query.asciidoc]: I think there's an extra space and an extra 'so' here.

File: docs/reference/query-dsl/function-score-query.asciidoc [hunk 3]
```diff
@@ -305,10 +305,14 @@ There are a number of options for the `field_value_factor` function:
...
+          between 0 and 1, so it is recommended to use `log1p` instead.
 | `log1p` | Add 1 to the field value and take the common logarithm
 | `log2p` | Add 2 to the field value and take the common logarithm
-| `ln` | Take the https://en.wikipedia.org/wiki/Natural_logarithm[natural logarithm] of the field value
+| `ln` | Take the https://en.wikipedia.org/wiki/Natural_logarithm[natural logarithm] of the field value.
+         Note: this function will return a negative value and cause an error if used on values
+         between 0 and 1,so it is recommended to use `ln1p` instead.
```
Comment (jtibshirani) [inline docs/reference/query-dsl/function-score-query.asciidoc]: Missing space here.

**Review by jtibshirani — APPROVED** [review]
Thanks @gwbrown, this looks good to me.

File: docs/reference/query-dsl/function-score-query.asciidoc (hunk 1 above)
Comment (gwbrown) [reply docs/reference/query-dsl/function-score-query.asciidoc]: Good call, I've changed these to `NOTE`.

**Comment by gwbrown** [comment]
Thanks for the review @jtibshirani!

**Pull request merged**